tc/src/main/webapp/WEB-INF/templates/coursesTable.html

```diff
@@ -0,0 +1,56 @@
+<div class="search-box">
+    <form class="form-search" th:action="@{/} + ${path}" method="get" >
+        <select class="span3" name="filterOption">
+            <option th:selected="All" th:text="All"></option>
+            <option th:each="category : ${categories}"
+                    th:text="${category.name}"
+                    th:selected="${categories.contains(category.name)}">Options</option>
+        </select>
+        <button class="btn" type="submit">Apply</button>
+    </form>
+</div>
+<table class="table table-striped table-bordered">
+    <thead>
+        <tr>
+            <th class="span1">Id</th>
+            <th>Course</th>
+            <th sec:authorize="hasAuthority('User')" class="span3">Category</th>
+            <th sec:authorize="hasAuthority('User')" class="span1">S/A</th>
+            <th sec:authorize="hasAuthority('User')" class="span1">Grade</th>
+            <th class="span2">Action</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr th:each="course: ${courses}">
+            <td th:text="${course.id}">Id</td>
+            <td>
+                <a th:href="@{/courses/} + ${course.id}" th:text="${course.name}">Course </a>
+            </td>
+            <td sec:authorize="hasAuthority('User')">
+                <div th:text="${course.category.name}"></div>
+            </td>
+            <td sec:authorize="hasAuthority('Lector')">
+                <a class="btn btn-mini" th:href="@{/courses/} + ${course.id}+ @{/update}">Update</a>
+            </td>
+            <td sec:authorize="hasAuthority('User')">
+                <a th:href="@{'/courses/'+${course.getId()}+ '/participants'}"
+                   th:text="${course.getSubscribers().size()>0?course.getSubscribers().size():'0'}
```

Comment:
getSubscribers() return set. This set is join table and can be empty, but not null.
check i will edit